addons/InfoBar/InfoBar.lua

```diff
@@ -0,0 +1,303 @@
+    * Neither the name of InfoBar nor the
+      names of its contributors may be used to endorse or promote products
+      derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL KENSHI BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.]]
+
+_addon.name = 'Infobar'
+_addon.author = 'Kenshi'
+_addon.version = '1.0'
+_addon.commands = {'ib', 'infobar'}
+
+config = require('config')
+texts = require('texts')
+require('vectors')
+res = require('resources')
+require('sqlite3')
+
+defaults = {}
+defaults.NoTarget = "${name} (${main_job}${main_job_level}/${sub_job}${sub_job_level}) (${x},${y},${z})"
+defaults.TargetPC = "${name}"
+defaults.TargetNPC = "${name}"
+defaults.TargetMOB = "${name}"
+defaults.display = {}
+defaults.display.pos = {}
+defaults.display.pos.x = 0
+defaults.display.pos.y = 0
+defaults.display.bg = {}
+defaults.display.bg.red = 0
+defaults.display.bg.green = 0
+defaults.display.bg.blue = 0
+defaults.display.bg.alpha = 102
+defaults.display.text = {}
+defaults.display.text.font = 'Consolas'
+defaults.display.text.red = 255
+defaults.display.text.green = 255
+defaults.display.text.blue = 255
+defaults.display.text.alpha = 255
+defaults.display.text.size = 12
+
+settings = config.load(defaults)
+
+box = texts.new("", settings.display, settings)
+
+local infobar = {}
+infobar.new_line = '\n'
+
+windower.register_event('load',function()
+    db = sqlite3.open(windower.addon_path..'\database.db')
+    notesdb = sqlite3.open(windower.addon_path..'/notes.db')
+    notesdb:exec('CREATE TABLE IF NOT EXISTS notes(name TEXT primary key, note TEXT)')
+    if not windower.ffxi.get_info().logged_in then return end
+    local target = windower.ffxi.get_mob_by_target('st') or windower.ffxi.get_mob_by_target('t') or windower.ffxi.get_player()
+    get_target(target.index)
+end)
+
+windower.register_event('unload',function()
+    db:close()
+    notesdb:close()
+end)
+
+function getDegrees()
+    local target = windower.ffxi.get_mob_by_target('st') or windower.ffxi.get_mob_by_target('t') or windower.ffxi.get_mob_by_target('me')
+    local target_heading = V{}.from_radian(target.facing)
+    local angleInDegrees = (math.atan2(target_heading[1], target_heading[2]) * (180 / math.pi))
+    local Degrees = angleInDegrees % 360
+    return math.floor(Degrees)
+end
```

Comment:
This function can be drastically simplified:
```lua
function getDegrees(value)
    return math.round(360 / math.tau * value)
end
```

Yours returns north for 0°, but the game uses 0° for east. I don't know how InfoBar displayed degrees by value (if it used 0° for north or east or SSW or something), but I think it would make sense to part with convention in favor of game-compatibility, since it doesn't really matter where 0° points to, so it may as well match what the game has.

Just a suggestion though, if you think it makes sense to make 0° be north, then use this:
```lua
function getDegrees(value)
    return math.round(360 / math.tau * (value - math.pi / 2))
end
```

Note that I also changed the signature to get the `facing` value passed in. The `get_mob_by_x` functions are pretty expensive and you do it multiple times on every prerender. I'd suggest having it done once in the calling function (which you already have, the `pos` object), then passing it into both this function and the next one.